apis/rust/node/src/node/mod.rs

```diff
@@ -157,10 +158,25 @@ impl DoraNode {
             drop_stream,
             cache: VecDeque::new(),
             dataflow_descriptor,
+            warned_unknown_output: BTreeSet::new(),
         };
         Ok((node, event_stream))
     }
 
+    fn validate_output(&mut self, output_id: &DataId) -> bool {
+        if !self.node_config.outputs.contains(output_id) {
+            if self.warned_unknown_output.contains(output_id) {
+                warn!("Output {output_id} is not part of the node's output list.");
```

Comment:
I'm not sure if I understand this logic here? We don't warn on the first attempt, but on all subsequent attempts?

Reply:
My bad sorry!